### DISCUSSION TITLE
Facing issues while creating a route to consul

### DISCUSSION THREAD

**Discussion by suniljangir-datdyn** [apache/apisix#9759] [Q&A]
I have successfully set up APISIX in Docker, along with Consul.
To integrate Consul with APISIX, I followed the instructions provided in this article: (https://apisix.apache.org/blog/2022/02/25/consul-api-gateway/).

So far, I have achieved the following:

1) I can access both APISIX and Consul through their respective ports i.e 9200 and 8500, in a web browser.
2) I have registered some test service to Consul by making an API call as described in step 9 of the article.
3) When I execute the command "$ curl http://127.0.0.1:8500/v1/kv/upstreams/webpages?keys," got a 200 response containing the keys for the services I registered.

Everything appears to be functioning correctly up to this point.

However, I encountered an issue when attempting to create a route and enable Consul, following the instructions provided in the article, I sent the following request:

--------------------
curl http://127.0.0.1:9080/apisix/admin/routes -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X POST -d '
{
    "uri": "/consul/*",
    "upstream": {  
        "service_name": "http://consul-server1:8500/v1/kv/upstreams/webpages/",
        "type": "roundrobin",
        "discovery_type": "consul_kv"
    }
}'
-----------------------

Unfortunately, my route was not created, and I received a 404 response with the error message: "404 Route Not Found."

I have attached the Consul configuration from apisix_conf/config.yml and the docker-compose file for reference. (file extension is changed to .txt as I can't upload YML files here directly.)

If anyone encountered the same, please have a look and suggest how can I solve this.

P.S. - I've tried service_name as "upstreams", "upstreams/webpages" and "webpages" as some websites had stated that you just need to put the service name only, instead of the URL. But nothing changed.

[config.yaml.txt](https://github.com/apache/apisix/files/11905294/config.yaml.txt)
[docker-compose.yml_temp1.txt](https://github.com/apache/apisix/files/11905296/docker-compose.yml_temp1.txt)
[docker-compose.yml_temp.txt](https://github.com/apache/apisix/files/11905297/docker-compose.yml_temp.txt)

**Comment by Revolyssup** [comment]
@suniljangir-datdyn Can you try using a separate port?
You can add this https://github.com/apache/apisix/blob/f2337f25d4bdbe7291810f36c0fbedd5cea8d29d/conf/config-default.yaml#L620 to your config and try again on port 9180. You have already mapped port 9180 on your container in the docker-compose so it should work.
